File: benchmarks/cross-language/fibonacci/bench.cs
using System;

class Program
{
    static long FibRecursive(long n)
    {
        if (n <= 1) return n;
        return FibRecursive(n - 1) + FibRecursive(n - 2);
    }

    static long FibIterative(long n)
    {
        if (n <= 1) return n;
        long a = 0, b = 1;
        for (long i = 2; i <= n; i++)
        {
            long temp = a + b;
            a = b;
            b = temp;
        }
        return b;
    }

    static void Main()
    {
        // Recursive (expensive)
        long result = FibRecursive(30);
        Console.WriteLine(result);
        // Iterative (fast)
        for (int i = 0; i < 100_000; i++)
        {
            FibIterative(30);
        }
        Console.WriteLine("done");
    }
}

File: benchmarks/cross-language/fibonacci/bench.py
def fib_recursive(n: int) -> int:
    if n <= 1:
        return n
    return fib_recursive(n - 1) + fib_recursive(n - 2)

def fib_iterative(n: int) -> int:
    if n <= 1:
        return n
    a = 0
    b = 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b

def main():
    # Recursive (expensive)
    result = fib_recursive(30)
    print(result)
    # Iterative (fast)
    for _ in range(100_000):
        fib_iterative(30)
    print("done")

if __name__ == "__main__":
    main()
